### DISCUSSION TITLE
Quais os objetivos reais do projeto?

### DISCUSSION THREAD

**Discussion by RodrigoCSoares** [BrasilAPI/BrasilAPI#241] [Q&A]
Olá, quando vi o título de "Vamos transformar o Brasil em uma API?" achei algo bem amplo, porém com muito potencial e que me deixou entusiasmado em colaborar. Mas ao ler o README e as issues fiquei um pouco perdido, aparentemente o único endpoint implementado foi o de busca de CEPs, que basicamente faz uma chamada da API do cep-promise. Sendo esse endpoint o único citado no README do projeto. Então ficaram alguns questionamentos na minha cabeça:

1. Quais são os reais objetivos do projeto? 
2. O que seria "transformar o Brasil em uma API"? 
3. Existe um roadmap? 
4. Será que essas informações não deveriam estar mais acessíveis?

**Comment by wilsonneto-swe** [comment]
O @filipedeschamps  quem vai falar melhor, mas vejo o objetivo como sendo ser um agregador/centralizador de apis úteis e importantes do Brasil, sendo além de um agregador uma interface e um cache, seguindo o conceito de stale while revalidate.

**Comment by xteoos** [comment]
Acredito que o CEP seja só o começo

**Comment by MateusGabi** [comment]
Olá, @RodrigoCSoares, @wilsonneto-dev e @xteoos 

O @filipedeschamps falou um pouco sobre isso no https://www.youtube.com/watch?v=WVtijTZPZck

Resumo:
- Uma API que possui utilidades sobre o Brasil. Um CEP é exemplo, mas poderia ser feito algo como: dado uma cidade, retorna o nome do prefeito, vereadores, etc.
- Formatação de dinheiro?
- Ações da Bovespa em Real-time
- Pode ter qualquer coisa sobre o Brasil

**Comment by RodrigoCSoares** [comment]
@MateusGabi @wilsonneto-dev Eu vi os vídeos e o que o @filipedeschamps comentou sobre o assunto, por isso mesmo achei muito amplo. Esse issue seria exatamente para tentar entender melhor qual seria o próximo passo, qual o próximo endpoint seria interessante implementar? Porque acredito que por enquanto as definições do projeto estejam muito vagas o que dificulta um pouco a colaboração e acaba ficando sem um ponto focal de implementações e discussões. Acredito que um roadmap ou uma definição mais específica no README ajudariam nessas questões.

 Vocês concordam com essa visão?

**Comment by JeiselColaco** [comment]
@RodrigoCSoares, concordo com sua posição. Se não criar um roadmap para que seja possível direcionar os esforços acaba por dispersar por demais as iniciativas....

**Comment by giovannicruz97** [comment]
Eu achei muito legal os comentários todos. Acho que esse projeto de centralizar os dados sobre o Brasil tem um potencial muito grande de facilitar processos que hoje são manuais em orgãos públicos e até de empresas que dependem de pessoas para realizarem operações manuais. Esse projeto tem um leque muito grande. Acho que realmente precisamos de um roadmap para saber onde queremos guiar o projeto. Claro, isso daqui vai ser gigante uma hora! (se tudo ocorrer bem), mas esse energia precisa ser canalizada.

**Comment by schirrel** [comment]
Achei muito interessante a ideia também.
Porém me resta uma duvida desde que vi o projeto, e reforçado com algumas issues que vi, e até com respostas aqui.
São dois exemplos:
> " mas poderia ser feito algo como: dado uma cidade, retorna o nome do prefeito, vereadores, etc."
> "que hoje são manuais em orgãos públicos e até de empresas que dependem de pessoas para realizarem operações manuais."
Ou na issues como informações de bairros por exemplo.

Muitas dessas informações citadas realmente não existem origens online, a não ser em site estáticos, antigos ou ainda públicos, esses sem apis também e que sabemos que ate para abrir uma API o processo pode ser burocrático. Estou falando num contexto geral, abrangendo cidades pequenas, ou em estados menos 'tecnológicos', informações que estão em entidades que ainda não fazem uso da tecnologia tanto quanto grandes centros e polos bem desenvolvidos.

Pensando nesses lugares e as informações que eles tem, não seria interessante, ter o BrasilAPI como um repositório de informações "open source"? Onde moradores/devs/pessoas que queiram dissipar a informação sobre o lugar em si, possam informar/validar/editar a informações sobre determinado lugar? Nem que seja uma extração de dados quase periódica de algum lugar (Uma coisa interessante também  é a issue #13 )

Claro que a ideia do Filipe já é 'fantardiga' já por si só, mas muitas vezes nos devs temos o defeito de olhar tudo com olhos tecnológicos o que não é a realidade do nosso brasil.

**Comment by RodriAndreotti** [comment]
> 
> 
> Achei muito interessante a ideia também.
> Porém me resta uma duvida desde que vi o projeto, e reforçado com algumas issues que vi, e até com respostas aqui.
> São dois exemplos:
> 
> > " mas poderia ser feito algo como: dado uma cidade, retorna o nome do prefeito, vereadores, etc."
> > "que hoje são manuais em orgãos públicos e até de empresas que dependem de pessoas para realizarem operações manuais."
> > Ou na issues como informações de bairros por exemplo.
> 
> Muitas dessas informações citadas realmente não existem origens online, a não ser em site estáticos, antigos ou ainda públicos, esses sem apis também e que sabemos que ate para abrir uma API o processo pode ser burocrático. Estou falando num contexto geral, abrangendo cidades pequenas, ou em estados menos 'tecnológicos', informações que estão em entidades que ainda não fazem uso da tecnologia tanto quanto grandes centros e polos bem desenvolvidos.
> 
> Pensando nesses lugares e as informações que eles tem, não seria interessante, ter o BrasilAPI como um repositório de informações "open source"? Onde moradores/devs/pessoas que queiram dissipar a informação sobre o lugar em si, possam informar/validar/editar a informações sobre determinado lugar? Nem que seja uma extração de dados quase periódica de algum lugar (Uma coisa interessante também é a issue #13 )
> 
> Claro que a ideia do Filipe já é 'fantardiga' já por si só, mas muitas vezes nos devs temos o defeito de olhar tudo com olhos tecnológicos o que não é a realidade do nosso brasil.

Estava vasculhando umas APIs para responder outra issue aqui e achei isso, que talvez possa ajudar nessa questão:
[http://www.ipeadata.gov.br/Default.aspx](http://www.ipeadata.gov.br/Default.aspx)

Aqui é possível acessar dados do TSE e ter várias informações referentes às eleições, o que pode ajudar a obter os dados de prefeito, vereadores etc.

**Comment by AldairAvelino** [comment]
Eu acredito de que os objetivos reais desse  projeto é tornar a vida de todos os programadores muito mais **simplificada**.
Tornando possíveis diversos tipos de **integrações** em seus projetos pessoais e não só.
